r/configure

```diff
@@ -173,6 +186,11 @@ else
         BUNDLED_LIBS=`echo "$BUNDLED_LIBS" | sed -e "s/\\.a lib/ -l/g" | sed -e "s/\\.a$//" | sed -e "s/^lib/-l/" | tr '\n' ' ' | sed -e "s/ $//"`
         PKG_DIRS="-L`pwd`/$LIB_DIR"
 
+        # Use pkg-config to do static linking of libarrow's dependencies
+        if [[ "$ARROW_DEPENDENCY_SOURCE" = "AUTO" || "$ARROW_DEPENDENCY_SOURCE" = "SYSTEM" ]]; then
```

Comment:
We should use `[` not `[[` because `/bin/sh` may not be bash`: https://github.com/ursacomputing/crossbow/runs/3280854203?check_suite_focus=true#step:7:4328

```text
./configure: 190: [[: not found
./configure: 190: SYSTEM: not found
```